src/main/java/com/jcabi/log/VerboseProcess.java

```diff
@@ -473,6 +486,87 @@ public Void call() throws Exception {
             }
             return null;
         }
+        
+        private void logFromReader(final BufferedReader reader,
+                final BufferedWriter writer) throws IOException, 
+                ClosedByInterruptException {
+            StringBuilder builder = new StringBuilder();
+            String previous = NULL_STRING;
+            int lineCount = 0;
+            while (true) {
+                if (Thread.interrupted()) {
+                    Logger.debug(
+                        VerboseProcess.class,
+                        "explicitly interrupting read from buffer"
+                    );
+                    break;
+                }
+                if (previous != NULL_STRING) {
+                    builder.append(previous).append(NEW_LINE);
+                    previous = NULL_STRING;
+                }
+                final String line = reader.readLine();
+                if (line == null) {
+                    doLog(writer, this.level, builder);
+                    break;
+                }
+                if (shouldAppend(line)
+                        && (++lineCount < MAX_STACK_LENGTH)) {
+                    builder.append(line).append(NEW_LINE);
+                } else {
+                    if (builder.length() > 0) {
+                        doLog(writer, this.level, builder);
+                        builder = new StringBuilder();
+                    }
+                    lineCount = 1;
+                    previous = line;
+                }
+            }
+        }
+
+        /**
+         * Logs supplied StringBuilder to supplied Logger and Writer.
+         * @param writer Writer to use
+         * @param level Level to log at
+         * @param builder StringBuilder with log statement
+         * @throws IOException writer could throw this
+         */
+        private static void doLog(final BufferedWriter writer, final Level level,
```

Comment:
@dean-e-clark one character too long, let's move `final Level level,` to a new line